pkg/api/api.go

```diff
@@ -73,7 +73,8 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Get("/plugins/:id/", reqSignedIn, hs.Index)
 	r.Get("/plugins/:id/edit", reqSignedIn, hs.Index) // deprecated
 	r.Get("/plugins/:id/page/:page", reqSignedIn, hs.Index)
-	r.Get("/a/:id/*", reqSignedIn, hs.Index) // App Root Page
+	r.Get("/a/:id/", reqSignedIn, middleware.ReqAppRoles, hs.Index)  // App root page
+	r.Get("/a/:id/*", reqSignedIn, middleware.ReqAppRoles, hs.Index) // App sub pages
```

Comment:
I had to duplicate the route definition for `/` and `/*` because the latter is mandatory (not empty `*` is allowed), so the `/` is needed to support those applications that only uses the root page (i.e. tabs-based app).

Note that this pattern (defining both `/` and `/*`) is not something new, but something already present multiple times in the same file.